compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/extended/MembarNode.java

```diff
@@ -67,6 +78,30 @@ public void generate(NodeLIRBuilderTool generator) {
         generator.getLIRGeneratorTool().emitMembar(barriers);
     }
 
+    public int getBarriers() {
+        return barriers;
+    }
+
+    public void setAccess(ValueNode access) {
+        Node old = this.access;
```

Comment:
The typical pattern for those is
```
updateUsages(this.access, access);
this.access = access
```
I don't think it should make a difference but it might be better to stick to the common pattern.

Reply:
Thanks for the tip - I'll try that and see if it helps.

Reply:
Well, that didn't make any difference :-)